Bug: compile error with gcc

Error "template instantiation depth exceeds maximum of 300" is reported with g++ compiler.
I'm not sure why clang compiler not occurs this error though.

### How to reproduce
```sh
user@localhost:~work/crc32-11$ mkdir build
user@localhost:~work/crc32-11$ cd build
user@localhost:~work/crc32-11/build$ cmake ..
user@localhost:~work/crc32-11/build$ make
```

```
[ 50%] Building CXX object CMakeFiles/crc32++11.dir/src/main.cpp.o
In file included from /home/takuya/work/crc32-11/src/main.cpp:28:0:
/home/takuya/work/crc32-11/src/crc32.hpp: In instantiation of 'constexpr const uint32_t crc32_table_generator<79764919u>::crc32_table_elem<111u, 1u>::value':
/home/takuya/work/crc32-11/src/crc32.hpp:52:41:   recursively required from 'constexpr const uint32_t crc32_table_generator<79764919u>::crc32_table_elem<111u, 7u>::value'
/home/takuya/work/crc32-11/src/crc32.hpp:62:25:   recursively required from 'constexpr const auto crc32_table_generator<79764919u>::array_impl<255ul>::value'
/home/takuya/work/crc32-11/src/crc32.hpp:62:25:   required from 'struct crc32_table_generator<79764919u>::array_impl<255ul>'
/home/takuya/work/crc32-11/src/crc32.hpp:72:45:   required from 'constexpr const std::array<unsigned int, 256ul> crc32_table_generator<79764919u>::value'
/home/takuya/work/crc32-11/src/crc32.hpp:83:63:   required from 'uint32_t crc32(uint32_t, iterator_t, iterator_t) [with unsigned int poly = 79764919u; iterator_t = std::istream_iterator<unsigned char>; uint32_t = unsigned int]'
/home/takuya/work/crc32-11/src/main.cpp:39:56:   required from here
/home/takuya/work/crc32-11/src/crc32.hpp:52:41: fatal error: template instantiation depth exceeds maximum of 300 (use -ftemplate-depth= to increase the maximum)
    (crc32_table_elem<index, N-1>::value << 1) ^ denominator<uint32_t, carry>::value;
                                         ^
compilation terminated.
CMakeFiles/crc32++11.dir/build.make:62: recipe for target 'CMakeFiles/crc32++11.dir/src/main.cpp.o' failed
make[2]: *** [CMakeFiles/crc32++11.dir/src/main.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/crc32++11.dir/all' failed
make[1]: *** [CMakeFiles/crc32++11.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8)
 project(crc32++11 CXX)
 
-set(CMAKE_CXX_FLAGS "-std=c++11 -W -Wall -ftemplate-depth=300")
+set(CMAKE_CXX_FLAGS "-std=c++11 -W -Wall -ftemplate-depth=600")
 set(CMAKE_CXX_FLAGS_DEBUG "-O2 -g")
 set(CMAKE_CXX_FLAGS_RELEASE "-O2")
 

